### ISSUE TITLE
Expo 31.0 and language-detector.js

### ISSUE THREAD

**Issue by borgogelli** [ashour/react-native-i18n-demo#1]
`The syntax "import Expo from 'expo'" has been deprecated in favor of "import { A, B, C } from 'expo'" or "import * as Expo from 'expo'". This sets us up to support static analysis tools like TypeScript and dead-import elimination better in the future. The deprecated import syntax will be removed in SDK 32.`

**Comment by borgogelli** [comment]
`Expo.DangerZone.Localization.getCurrentLocaleAsync() is deprecated. Use `Expo.Localization.locale` instead.`

**Comment by ashour** [comment]
Thanks @borgogelli. I'll address this ASAP.

**Comment by pckz** [comment]
Thanks guys. I'm having two issues with a published production app (works in expo). I did the implementation based in this demo.

1. How would it be the code changing Expo.DangerZone.Localization.getCurrentLocaleAsync() and using Expo.Localization.locale instead? What changes I have to do?
2. In file `config/i18n.js` I get syntax error in this line:
```
export const supportedLocales = {
    en: {
        name: "English",
        translationFileLoader: () => require('../lang/en.json'),
        momentLocaleLoader: () => Promise.resolve(),
    },
    ar: {
        name: "Español",
        translationFileLoader: () => require('../lang/es.json'),
        momentLocaleLoader: () => import('moment/locale/es'), // <-- in this line
    },
};
```
I get: unexpected token.
So instead I used `momentLocaleLoader: () => Promise.resolve(),` too. Works, but the production app fails for some reason.
Thanks for your advices, it is very appreciated.

**Comment by ashour** [comment]
Hey @pckz sorry about the late reply. I didn't get an email notification and will change my notification settings to address this. I suppose because your app is a production app that you've solved the issue or worked around it already. I'm maintaining quite a few repos/articles, and I have yet to get to this one. It's still on my radar, so I will get to it eventually. In the meantime, did you find a solution?

**Comment by kubapyla** [comment]
@ashour Hello, I'd like to mention that currently, using exact code from the repo it partially works (I've had to change `Expo.DangerZone.Localization.getCurrentLocaleAsync() ` to `Localization.getLocalizationAsync().then(obj => {callback(obj.locale.split('_')[0])})` and added babel dynamic imports (I think that part is not working properly as I get warnings (`Cannot read property 'translationFileLoader' of undefined`)

but `i18n.init().then(() => ...)` is not executing at all, so the layout direction is not changing, translation strings are working fine.

**Comment by ashour** [comment]
Hey @kubapyla, I'm updating the app for Expo 32 and taking care of the `DangerZone` deprecation right now. I think that Expo changed the format of its returned locale. So `Expo.Localization.locale` returns something like `"en-US"` not `"en_US"` (it's a dash instead of an underscore). So the callback should take something like `Expo.Localization.locale.split('-')[0]`.

**Comment by kubapyla** [comment]
Just wanted to say that's a really great job, at least a proper clever implementation of translations with RTL support in React Native! Everything is working smoothly. Not so much so good content on the web about that topic :)

**Comment by ashour** [comment]
You're quite welcome @kubapyla. I'm glad it's been useful to you :)

**Comment by soryboums** [comment]
Hello @ashour , I just started on RN and I found your repo by reading your fantastic tutorial here https://phraseapp.com/blog/posts/react-native-i18n-with-expo-and-i18next-part-1/.
I tried to apply the same logic of implementation on my project and I have notice one thing: 
When the configured locale on the phone is not defined in supportedLocales list (i18n.js file) the application will not load. The issue is that we are not defaulting to the fallback language while loading the translation and initiating momentjs.

Step to reproduce:
- Change your phone language and use French for exemple.
- Launch the appilcation

Result: 
- App will stock and loading page.

I can do a PR if you accept the issue :)
